include/sg14/bits/fixed_point_math.h

```diff
@@ -0,0 +1,142 @@
+
+//          Copyright Timo Alho 2016.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file ../LICENSE_1_0.txt or copy at
+//          http://www.boost.org/LICENSE_1_0.txt)
+
+/// \file
+/// \brief some cmath specializations for `sg14::fixed_point` type;
+
+#ifndef FIXED_POINT_MATH_H_
+#define FIXED_POINT_MATH_H_
+
+#include <sg14/fixed_point.h>
+#include <type_traits>
+
+namespace sg14 {
+
+    namespace _fixed_point_impl {
+
+        template<class FixedPoint>
+        constexpr FixedPoint rounding_conversion(double d) {
+            using one_longer = make_fixed<FixedPoint::integer_digits, FixedPoint::fractional_digits + 1>;
```

Comment:
Does this function do more than increase accuracy by a bit? As it invariably promotes an intermediate type to the next size up, it precludes larger types. Rounding is necessarily a problem when using built-in types as `Exp` so trying to fix it for this one function may not be a worthwhile trade-off overall.